sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -168,32 +189,45 @@ def run_query(query, args, pipeline_options):
       result.cancel()
     else:
       result.wait_until_finish()
-  except NotImplementedError as exp:
-    print(exp)
+  except Exception as exc:
+    thread_errors.append(str(exc))
 
 
 def run(argv=None):
   args, _ = parse_args()
 
   queries = {
       0: query0,
-      # TODO(mariagh): Add more queries
+      1: query1,
+      2: query2,
+      # TODO(mariagh): query4
   }
 
   for i in args.query:
     args, pipeline_options = parse_args()
     logging.info('Running query %d', i)
 
-    job_setup_time = 20 * 1000    # in milliseconds
-    query_duration = pipeline_options.view_as(TestOptions).wait_until_finish_duration + job_setup_time # pylint: disable=line-too-long
-    command = Command(run_query, args=[queries[i], args, pipeline_options])
-    try:
-      command.run(timeout=query_duration/1000)
-    except Exception as exp:
-      logging.error('Command failed with "%s"', exp)
-
-  logging.info('Queries run successfully: %s', args.query)
-
+    # The DirectRunner is the default runner, and it needs
+    # special handling to cancel streaming jobs.
+    launch_from_direct_runner = pipeline_options.view_as(
+        StandardOptions).runner in [None, 'DirectRunner']
+
+    if launch_from_direct_runner:
+      thread_errors = collections.defaultdict(list)
+      command = Command(run_query, args=[queries[i], args, pipeline_options,
+                                         thread_errors[i]])
+      query_duration = pipeline_options.view_as(TestOptions).wait_until_finish_duration # pylint: disable=line-too-long
+      command.run(timeout=query_duration / 1000)
+      if thread_errors[i]:
```

Comment:
Do you mean if thread_errors contains a thing?

Based on the thread code above I think it won't have more 1 item.

Reply:
No. As explained above, `thread_errors` contains exceptions from all of the queries. I realized that the name is confusing and renamed it to be `query_errors`